[BUG] Unreleased OpenSearch integration tests are failing to build an instance

### What is the bug?

The `Integration with Unreleased OpenSearch / test (main, 11, 17)` test is failing in what looks like an unrelated issue:

```
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Build file '/home/runner/work/opensearch-php/opensearch-php/opensearch/buildSrc/build.gradle' line: 78

* What went wrong:
A problem occurred evaluating project ':buildSrc'.
> At least Java 21 is required to build opensearch gradle tools
```

### How can one reproduce the bug?
_Steps to reproduce the behavior._

### What is the expected behavior?
_A clear and concise description of what you expected to happen._

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

--- a/.github/workflows/test_unreleased.yml
+++ b/.github/workflows/test_unreleased.yml
@@ -15,9 +15,9 @@ jobs:
       fail-fast: false
       matrix:
         entry:
-          - { opensearch_ref: "1.x", java: 11 }
-          - { opensearch_ref: "2.x", java: 11, gradle-runtime: 17 }
-          - { opensearch_ref: "main", java: 11, gradle-runtime: 17 }
+          - { opensearch_ref: "1.x", java: 21 }
+          - { opensearch_ref: "2.x", java: 21, gradle-runtime: 21 }
+          - { opensearch_ref: "main", java: 21, gradle-runtime: 21 }
     steps:
       - name: Checkout PHP Client
         uses: actions/checkout@v4

